cmd/eksctl-anywhere/cmd/createcluster.go

```diff
@@ -167,6 +167,13 @@ func (cc *createClusterOptions) createCluster(cmd *cobra.Command, _ []string) er
 			CreateBootstrapClusterOptions: deps.Provider,
 		}
 		wflw.WithHookRegistrar(awsiamauth.NewHookRegistrar(deps.AwsIamAuth, clusterSpec))
+
+		// Not all provider implementations want to bind hooks so we explicitly check if they
+		// want to bind hooks before registering it.
+		if registrar, ok := deps.Provider.(management.CreateClusterHookRegistrar); ok {
```

Comment:
Is this to avoid the no-op methods in providers?

Reply:
Yes.